test/contrib/rails/rails_sidekiq_test.rb

```diff
@@ -73,4 +73,35 @@ def perform(); end
     assert_equal('agent1.example.com', @tracer.writer.transport.hostname)
     assert_equal('7777', @tracer.writer.transport.port)
   end
+
+  test 'Sidekiq middleware sends spans with the correct metadata' do
+    # configure Rails
+    update_config(:sidekiq_service, 'rails-sidekiq')
+
+    # add Sidekiq middleware
+    Sidekiq::Testing.server_middleware do |chain|
+      chain.add(Datadog::Contrib::Sidekiq::Tracer, tracer: @tracer)
+    end
+
+    sleep(1)
```

Comment:
Is that really necessary ? Is there a synchronization issue, and could it be solved without a random delay ? This is how we get flaky tests.